### PR TITLE
Update Strap configuration

### PR DESCRIPTION
**Pull request by ricardobcl**
Fixes:
* Updated dependencies, which fixes the OAuth issue found when trying to authenticate with Github;
* Email retrieval bug was fixed;
* Render Error page instead of downloading it.

Updated:
* Strap bash script was updated from upstream, and the Command Line Tools installation was simplified;
* Flatten folder structure.

New:
* Add support for custom named repositories, other than 'dotfiles'.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -76,6 +78,8 @@ app
     try {
       await next();
     } catch (error) {
+      // make sure that the content-disposition is not set to attachment, to not download the error page
```
Comment (waldyrious) [inline app.js]:
```suggestion
      // Make sure that the content-disposition is not set to attachment,
      // to not download the error page.
```

**Comment by ricardobcl** [comment]
The first commits on this PR had the wrong author email and changing those commits broke the link between the original branch and this PR's reference of it.
Github offers no way to restore this link, so I have to create a new PR.

**Comment by ricardobcl** [comment]
The first commits on this PR had the wrong author email and changing those commits broke the link between the original branch and this PR's reference of it.
Github offers no way to restore this link, so I have to create a new PR.

**Comment by ricardobcl** [comment]
The first commits had the wrong author email and Github PR's can't handle changing this, without losing track of the original branch.

**Comment by ricardobcl** [comment]
The first commits had the wrong author email and Github PR's can't handle changing this, without losing track of the original branch.

**Comment by ricardobcl** [comment]
The first commits had the wrong author email and Github PR's can't handle changing this, without losing track of the original branch.

**Pull request closed without merging**